src/NATS.Client.Core/Internal/NatsUri.cs

```diff
@@ -65,7 +81,7 @@ public NatsUri CloneWith(string host, int? port = default)
 
     public override string ToString()
     {
-        return IsWebSocket && Uri.AbsolutePath != "/" ? Uri.ToString() : Uri.ToString().Trim('/');
+        return IsWebSocket && Uri.AbsolutePath != "/" ? _redactedUri.ToString() : _redactedUri.ToString().Trim('/');
```

Comment:
Suggestion: If the redacted `Uri` is only being used here, IMO we would be best of just making the right string for it based on other args for Ctor vs keeping a separate URI instance and `.ToString()`ing it every call.